### PR TITLE
Swagger changes for Stop and Resume protection APIs

### PR DESCRIPTION
**Pull request by pratikjoshi14**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json [hunk 1]
```diff
@@ -1520,6 +1520,345 @@
...
+            "in": "path",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "202": {
```
Comment (mentat9) [inline specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json:1556]: >202 [](http://example.com/codeflow?start=11&length=3)

Add x-ms-long-running-operation

File: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json [hunk 2]
```diff
@@ -1520,6 +1520,345 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataProtection/backupVaults/{vaultName}/backupInstances/{backupInstanceName}/resumeBackups": {
+      "post": {
```
Comment (mentat9) [inline specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json:1524]: >post [](http://example.com/codeflow?start=7&length=4)

Should add a description for each of the five new operations.

File: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json [hunk 3]
```diff
@@ -1520,6 +1520,345 @@
...
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "SuspendBackups": {
+            "$ref": "./examples/BackupInstanceOperations/SuspendBackups.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataProtection/backupVaults/{vaultName}/backupInstances/{backupInstanceName}/sync": {
```
Comment (mentat9) [inline specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json:1787]: >sync [](http://example.com/codeflow?start=169&length=4)

Sync is a pretty general action name. Adding a description will help, but maybe there's also a more descriptive action name?

File: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json [hunk 4]
```diff
@@ -4364,6 +4703,19 @@
...
+    "SyncBackupInstanceRequest": {
+      "description": "Sync BackupInstance Request",
+      "required": [
+        "forceResync"
+      ],
+      "type": "object",
+      "properties": {
+        "forceResync": {
+          "description": "Specifies if resync needs to be done forced otherwise only resync will happen for failure case.",
+          "type": "boolean"
```
Comment (mentat9) [inline specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json:4715]: >boolean [](http://example.com/codeflow?start=19&length=7)

ARM generally recommends against booleans in favor of string enums. Best option is to choose a better property (e.g. syncType: { "ForceResync", "LazySync", "Default" }). This type of property is easier to understand and use. Next best option if you really want a True/False value is to use a string enum "forceResync": { "True", "False" }. Both options allow adding values later without requiring a new API version.

Comment (pratikjoshi14) [reply specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json:4715]: Sure.. will fix.

File: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json (hunk 2 above)
Comment (pratikjoshi14) [reply specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json:1524]: Added.. Thanks.

File: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json (hunk 3 above)
Comment (pratikjoshi14) [reply specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json:1791]: Added description. This API is still in review mode and can rename before moving to stable version.

File: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json (hunk 4 above)
Comment (pratikjoshi14) [reply specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json]: Fixed. Thanks.

**Comment by pratikjoshi14** [comment]
@mentat9 / @weidongxu-microsoft : These changes are only in -preview version which is not live yet. Hence it should not break customers - as flagged in the PR.

File: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json [hunk 5]
```diff
@@ -1520,6 +1520,350 @@
...
+              "Azure-AsyncOperation": {
+                "description": "The URL of the resource used to check the status of the asynchronous operation.",
+                "type": "string"
+              },
+              "Retry-After": {
+                "description": "Suggested delay to check the status of the asynchronous operation. The value is an integer that represents the seconds.",
+                "type": "integer",
+                "format": "int32"
+              }
+            }
```
Comment (weidongxu-microsoft) [inline specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json:1573]: I don't think these headers need to be spelled out. As long as it is marked as `x-ms-long-running-operation`, all SDK will handle these.

On the other hand, is there empty response after LRO? If not, the response schema should be here.

**Review by weidongxu-microsoft — COMMENTED** [review]
@pratikjoshi14 

Is there any SDK released for this preview version?

The factor on live or not, SDK released or not is only factors to consider, you still need approval from Jeffrey.

File: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json (hunk 5 above)
Comment (sumitmal2711) [reply specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json:1573]: Yes the response body is empty

File: specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json (hunk 3 above)
Comment (sumitmal2711) [reply specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json:1791]: *preview. Thanks for the comment. We want to get the customer feedback so have released portal as well. We have noted renaming of the API and will take this as part of public preview when move to stable version in a month or so. As we will have to coordinate the change with our Portal team

**Review by sumitmal2711 — APPROVED** [review]

**Comment by sumitmal2711** [comment]
> @pratikjoshi14
> 
> Is there any SDK released for this preview version?
> 
> The factor on live or not, SDK released or not is only factors to consider, you still need approval from Jeffrey.

We published a powershell sdk for specific private preview customers who wanted to get early sneak peek. but no .net or other sdks were generated..

**Comment by pratikjoshi14** [comment]
> > @pratikjoshi14
> > Is there any SDK released for this preview version?
> > The factor on live or not, SDK released or not is only factors to consider, you still need approval from Jeffrey.
> 
> We published a powershell sdk for specific private preview customers who wanted to get early sneak peek. but no .net or other sdks were generated..

@JeffreyRichter : Can you please approve this PR?

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**